dataloader/tasks_segments_parallels.py

```diff
@@ -345,6 +360,57 @@ def load_parallels(json_parallels: [Parallel], db: StandardDatabase) -> None:
             print(f"Could not save parallel {parallel}. Error: ", e)
 
 
+
+def load_parallels_sorted(json_parallels: [Parallel], db: StandardDatabase,filename: str) -> None:
+    """
+    Given an array of parallel objects, load them all into the `sorted parallels` collection presorted.
+
+    :param json_parallels: Array of parallel objects to be loaded as-they-are.
+    :param db: ArangoDB connection object
+
+    """
+
+    db_collection_sorted = db.collection(COLLECTION_PARALLELS_SORTED_BY_FILE)
+    # I wonder if this can be done more efficiently, a lot of spaghetti code...    
+    parallels_sorted_by_src_position = sorted(json_parallels, key=lambda k: k['root_pos_beg'])
+    ids_sorted_by_src_position = []
+    for parallel in parallels_sorted_by_src_position:
+        ids_sorted_by_src_position.append(parallel['id'])
+        
+    parallels_sorted_by_tgt_position = sorted(json_parallels, key=lambda k: k['par_pos_beg'])                                           
+    ids_sorted_by_tgt_position = []
+    for parallel in parallels_sorted_by_tgt_position:
+        ids_sorted_by_tgt_position.append(parallel['id'])
+
+    parallels_sorted_by_length_par = sorted(json_parallels, key=lambda k: k['par_length'])                                          
+    ids_sorted_by_length_par = []
+    for parallel in parallels_sorted_by_length_par:
+        ids_sorted_by_length_par.append(parallel['id'])
+
+    parallels_sorted_by_length_root = sorted(json_parallels, key=lambda k: k['root_length'])
+    ids_sorted_by_length_root = []
+    for parallel in parallels_sorted_by_length_root:
+        ids_sorted_by_length_root.append(parallel['id'])
+    ids_sorted_by_length_par.reverse()
+    ids_sorted_by_length_root.reverse()
+
+    ids_sorted_random = ids_sorted_by_length_root
+    random.shuffle(ids_sorted_random)
```

Comment:
out of curiosity, why are we randomizing here?

Reply:
For the statistics (pie chart etc.) we cannot fetch all parallels from the database, but only the first 10000 (otherwise the query will take too long on long files). In order to guarantee that the statistics are representative, we randomize them before we fetch them with a limit of 10000. this avoids that we have a bias (for example when text X quotes text Y only in the first chapter and not in the second chapter, this would distort the statistics).